aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -39,6 +50,148 @@ fn is_string_type(context: &SafeNativeContext, type_arg: &Type) -> SafeNativeRes
     Ok(false)
 }
 
+/// Given the list of native function arguments, returns a tuple of its
+/// fields: (`aggregator id`, `value`, `limit`).
+pub fn get_aggregator_fields(
+    ty_arg: &Type,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<(AggregatorID, u128, u128)> {
+    match ty_arg {
+        Type::U128 => {
+            // Get aggregator information and a value to add.
+            let value = safely_pop_arg!(args, u128);
+            let (id, limit) = aggregator_value_as_u128(&safely_pop_arg!(args, StructRef))?;
+            Ok((id, value, limit))
+        },
+        Type::U64 => {
+            // Get aggregator information and a value to add.
+            let value = safely_pop_arg!(args, u64);
+            let (id, limit) = aggregator_value_as_u64(&safely_pop_arg!(args, StructRef))?;
+            Ok((id, value as u128, limit as u128))
+        },
+        _ => Err(SafeNativeError::Abort {
+            abort_code: EUNSUPPORTED_AGGREGATOR_TYPE,
+        }),
+    }
+}
+
+/***************************************************************************************************
+ * native fun create_aggregator<Element>(limit: Element): Aggregator<Element>;
+ **************************************************************************************************/
+
+fn native_create_aggregator(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert_eq!(args.len(), 1);
+
+    context.charge(AGGREGATOR_V2_CREATE_AGGREGATOR_BASE)?;
+    // Get the current aggregator data.
+    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+    let id = aggregator_data.generate_id();
+    let aggregator_id = AggregatorID::ephemeral(aggregator_data.generate_id());
+
+    match ty_args[0] {
+        Type::U128 => {
+            let limit = safely_pop_arg!(args, u128);
+            aggregator_data.create_new_aggregator(aggregator_id, limit);
+            Ok(smallvec![Value::struct_(Struct::pack(vec![
+                Value::u128(id as u128),
+                Value::u128(limit),
+            ]))])
+        },
+        Type::U64 => {
+            let limit = safely_pop_arg!(args, u64);
+            aggregator_data.create_new_aggregator(aggregator_id, limit as u128);
+            Ok(smallvec![Value::struct_(Struct::pack(vec![
+                Value::u64(id),
+                Value::u64(limit),
+            ]))])
+        },
```

Comment:
if you had pop_value_of_type function separately, as I suggested above, this can become:

```
    let limit = pop_value_of_type(args, ty_args[0]);
    aggregator_data.create_new_aggregator(aggregator_id, limit);
    match ty_args[0] {
        Type::U128 => Ok(smallvec![Value::struct_(Struct::pack(vec![
            Value::u128(id as u128),
            Value::u128(limit),
        ]))]),
        Type::U64 => Ok(smallvec![Value::struct_(Struct::pack(vec![
             Value::u64(id),
             Value::u64(limit as u64),
        ]))]),
        _ => Err(SafeNativeError::Abort {
            abort_code: EUNSUPPORTED_AGGREGATOR_TYPE,
        }),
    }

```

Reply:
Done.